### PR TITLE
chore: migrated smtpd to aiosmtpd

### PR DESCRIPTION
**Pull request by danielskinstad**
Smtpd is fully removed in python3.12.
Moved from smtpd and asyncore to aiosmtpd.

Ticket: QA-603

### CODE REVIEW

File: tests/requirements-python/python-requirements.in [hunk 1]
```diff
@@ -19,3 +19,4 @@ urllib3==1.26.18
 websockets==12.0
 flaky==3.7.0
 stripe==7.10.0
+aiosmtpd==1.4.4.post2
```
Comment (lluiscampos) [inline tests/requirements-python/python-requirements.in:22]: Nitpick: missing newline

File: testutils/infra/smtpd_mock.py [hunk 2]
```diff
@@ -11,45 +11,48 @@
...
+class SMTPServerMockHandler:
     def __init__(self, *args, **kwargs):
         self.messages = []
-        smtpd.SMTPServer.__init__(self, *args, **kwargs)
         self._msg_cond = Condition()
 
-    def process_message(self, peer, mailfrom, rcpttos, data, **kwargs):
-        self.messages.append(Message(peer, mailfrom, rcpttos, None, data))
-        logging.warning("got a message: %s" % mailfrom)
+    async def handle_DATA(self, server, session, envelope):
```
Comment (lluiscampos) [inline testutils/infra/smtpd_mock.py:46]: I suggest we add here a reference to the documentation:
* https://aiosmtpd.readthedocs.io/en/stable/handlers.html#handle_DATA

File: testutils/infra/smtpd_mock.py [hunk 3]
```diff
@@ -11,45 +11,48 @@
...
         self.peer = peer
-        self.mailfrom = mailfrom
+        self.mail_from = mail_from
         self.rcpttos = rcpttos
         self.subject = subject
         self.data = data
 
     def __repr__(self):
-        return f"<Message peer={self.peer} mailfrom={self.mailfrom} rcpttos={self.rcpttos} subject={self.subject}>"
+        return f"<Message peer={self.peer} mailfrom={self.mail_from} rcpttos={self.rcpttos} subject={self.subject}>"
```
Comment (lluiscampos) [inline testutils/infra/smtpd_mock.py:38]: Nice with the property rename. Please propagate it to the representation too:
```suggestion
        return f"<Message peer={self.peer} mail_from={self.mail_from} rcpttos={self.rcpttos} subject={self.subject}>"
```

File: testutils/infra/smtpd_mock.py [hunk 4]
```diff
@@ -11,45 +11,48 @@
...
-    def process_message(self, peer, mailfrom, rcpttos, data, **kwargs):
-        self.messages.append(Message(peer, mailfrom, rcpttos, None, data))
-        logging.warning("got a message: %s" % mailfrom)
+    async def handle_DATA(self, server, session, envelope):
+        peer = session.peer
+        mail_from = envelope.mail_from
+        rcpt_tos = envelope.rcpt_tos
+        data = envelope.content
+        self.messages.append(Message(peer, mail_from, rcpt_tos, None, data))
+        logging.warning("got a message: %s" % mail_from)
```
Comment (lluiscampos) [inline testutils/infra/smtpd_mock.py:52]: Should this really be a warning? I think we should downgrade to `info`

Comment (lluiscampos) [reply testutils/infra/smtpd_mock.py:52]: Also, please remove the `%` construction when modernizing code. For logging it is better with:
```suggestion
        logging.warning("got a message:", mail_from)
```

Comment (lluiscampos) [reply testutils/infra/smtpd_mock.py:52]: Finally, maybe this should be "got a message from:"

File: testutils/infra/smtpd_mock.py [hunk 5]
```diff
@@ -11,45 +11,48 @@
...
 
-    def process_message(self, peer, mailfrom, rcpttos, data, **kwargs):
-        self.messages.append(Message(peer, mailfrom, rcpttos, None, data))
-        logging.warning("got a message: %s" % mailfrom)
+    async def handle_DATA(self, server, session, envelope):
+        peer = session.peer
+        mail_from = envelope.mail_from
+        rcpt_tos = envelope.rcpt_tos
+        data = envelope.content
+        self.messages.append(Message(peer, mail_from, rcpt_tos, None, data))
```
Comment (lluiscampos) [inline testutils/infra/smtpd_mock.py:51]: Why are we declaring all these local variables here? In my opinion it is better to do
```suggestion
        self.messages.append(Message(session.peer, envelope.mail_from, envelope.rcpt_tos, None, envelope.content))
```

File: testutils/infra/smtpd_mock.py (hunk 3 above)
Comment (lluiscampos) [reply testutils/infra/smtpd_mock.py:38]: Now that I see the rest of the code, I suggest we also here rename `rcpttos` to `rcpt_tos`

**Review by lluiscampos — CHANGES_REQUESTED** [review]
It looks good! Thanks for doing this grunt work. I made some comments on minor things.

Remember that we need to add the dependency also in [here](https://github.com/mendersoftware/mender-test-containers/blob/master/backend-integration-testing/requirements-py3.txt) for the `backend-integration` tests to pass

**Comment by danielskinstad** [comment]
@mender-test-bot start pipeline

**Pull request merged**